### PR TITLE
texlive: 2019 -> 2020

### LINKED ISSUE #91994: TeXLive 2020 update
**Describe the bug**
TeXLive 2020 was released on 10 April 2020, Nix has only 2019 version at the moment.

- https://www.tug.org/texlive/

### Changes:

-    LuaHBTeX: lualatex is now based on LuaHBTeX, meaning that one can use the HarfBuzz renderer which in particular for complicated scripts (Tibetan, Bengali, …) works better than the Lua-based renderer. Note that luatex itself remains normal LuaTeX, only the luaLAtex one uses LuaHBTeX.
-    Versioned containers: this is a change under the hood we have been working on slowly over the last half year. Many distributions had problems with the changing content of our package containers (foobar.tar.xz while the name never changed. We have now changed all the infrastructure and TeX Live Manager to work with versioned containers foobar.rNNNNN.tar.xz. This should help quite some distributors!
-    Haranoaji (原ノ味): the default font for Japanese text was for long time the IPAex fonts, one of the few free fonts available. With 2020 we have switched to Haranoaji font family, which provides better support for JIS90/04 charsets, and more weights.

General:

-    The \input primitive in all TeX engines, including tex, now also accepts a group-delimited ﬁlename argument, as a system-dependent extension. The usage with a standard space/token-delimited ﬁlename is completely unchanged. The group-delimited argument was previously implemented in LuaTeX; now it is available in all engines. ASCII double quote characters (“) are removed from the ﬁlename, but it is otherwise left unchanged after tokenization. This does not currently aﬀect LaTeX’s \input command, as that is a macro redeﬁnition of the standard \input primitive.
-    New option –cnf-line for kpsewhich, tex, mf, and all other engines, to support arbitrary conﬁguration settings on the command line.
-    The addition of various primitives to various engines in this and previous years is intended to result in a common set of functionality available across all engines (LaTeX News #31).

- epTeX, eupTeX: New primitives \Uchar, \Ucharcat, \current(x)spacingmode, \ifincsname; revise \fontchar?? and \iffontchar. For eupTeX only: \currentcjktoken.

- LuaTeX: Integration with HarfBuzz library, available as new engines luahbtex (used for lualatex) and luajithbtex. New primitives: \eTeXgluestretchorder, \eTeXglueshrinkorder.

- pdfTeX: New primitive \pdfmajorversion; this merely changes the version number in the PDF output; it has no eﬀect on any PDF content. \pdfximage and similar now search for image ﬁles in the same way as \openin.

- pTeX: New primitives \ifjfont, \iftfont. Also in epTeX, upTeX, eupTeX.

- XeTeX: Fixes for \Umathchardef, \XeTeXinterchartoks, \pdfsavepos.

- Dvips: Output encodings for bitmap fonts, for better copy/paste capabilities (https://tug.org/TUGboat/tb40-2/tb125rokicki-type3search.pdf).

- MacTeX: MacTeX and x86_64-darwin now require 10.13 or higher (High Sierra, Mojave, and Catalina); x86_64-darwinlegacy supports 10.6 and newer. MacTeX is notarized and command line programs have hardened runtimes, as now required by Apple for install packages. BibDesk and TeX Live Utility are not in MacTeX because they are not notarized, but a README ﬁle lists urls where they can be obtained.

tlmgr and infrastructure:

-    Automatically retry (once) packages that fail to download.
-    New option tlmgr check texmfdbs, to to check consistency of ls-R ﬁles and !! speciﬁcations for each tree.
-    Use versioned ﬁlenames for the package containers, as in tlnet/archive/pkgname.rNNN.tar.xz; should be invisible to users, but a notable change in distribution.
-    catalogue-date information no longer propagated from the TeX Catalogue, since it was often unrelated to package updates.

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: texlive
# a list of nixos modules affected by the problem
module: texlive
```

cc @veprbl

### PR DESCRIPTION
**Pull request by veprbl**
###### Motivation for this change

Provide minor fixes for some items in `texlive.bin`. Update to the latest texlive snapshot.
Closes: #91994

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
```
nix path-info -h -S /nix/store/vb3l4zhkigrniqh400vl96qly8kpbhd1-texlive-combined-full-2020 -S /nix/store/0rlzarjqrplckcfdydk22na48dfj7zk9-texlive-combined-full-2019
/nix/store/0rlzarjqrplckcfdydk22na48dfj7zk9-texlive-combined-full-2019     3.4G
/nix/store/vb3l4zhkigrniqh400vl96qly8kpbhd1-texlive-combined-full-2020     3.8G
```
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by gebner** [comment]
Builds and works for me (tested on a few latex files).

**Comment by gebner** [comment]
This PR is still marked as draft, but pretty much LGTM from my side.  Is there anything that you still want to do?

**Comment by veprbl** [comment]
@gebner Just pushed a new changeset. Main difference is a change of default versioning scheme for unversioned packages, it now uses SVN revision number instead of the texlive year. There are also minor tweaks for luajit (cc @CrystalGamma for ppc64 changes).

We usually had major texlive updates sit for a while before getting merged, but we could try merging sooner this time.

**Comment by veprbl** [comment]
@GrahamcOfBorg build tests.texlive texlive.combined.scheme-full

**Comment by gebner** [comment]
> We usually had major texlive updates sit for a while before getting merged, but we could try merging sooner this time.

I know, in 2017 I merged the texlive update after it had been sitting around for over a month with no activity, and we only got bug reports after it was merged.

We don't gain much by letting this PR sit longer than necessary.  If it builds, then I think we should merge it.

**Comment by gebner** [comment]
I ran into one regression, the new biber 2.15 doesn't like authors such as `Jean{-}Marc`.  This is already fixed upstream (but not released yet): https://github.com/plk/biber/issues/329

File: pkgs/tools/typesetting/tex/texlive/bin.nix [hunk 1]
```diff
@@ -14,78 +14,26 @@
...
-      # Needed for ghostscript>=9.50
-      (fetchpatch {
-        name = "xdvipdfm-fix.patch";
-        url = "https://www.tug.org/svn/texlive/trunk/Build/source/texk/dvipdfm-x/spc_dvips.c?view=patch&r1=52765&r2=52764&pathrev=52765";
-        sha256 = "0qvrc7yxhbl5f4g340z8aql388bwib0m2gxd473skbmviy5bjr3f";
-        stripLen = 2;
-      })
-    ];
-
     # remove when removing synctex-missing-header.patch
```
Comment (dtzWill) [inline pkgs/tools/typesetting/tex/texlive/bin.nix:37]: Think this should've been removed before, per:comment?

Comment (veprbl) [reply pkgs/tools/typesetting/tex/texlive/bin.nix:37]: @dtzWill Nice catch, this should have been addressed back in update to 2019. Opened #101016. Thanks!

**Pull request merged**